### PR TITLE
mgr/dashboard: Add NFS status endpoint 

### PR DESCRIPTION
**Pull request by tspmelo**
Currently each time NFS page is opened and NFS Ganesha is not configured
an error notification is thrown and no extra information is given.
Now the user will be redirected to an information page.

Removed the orchestrator information since it no longer applies.

Fixes: http://tracker.ceph.com/issues/38399

Signed-off-by: Tiago Melo <tmelo@suse.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.html [hunk 1]
```diff
@@ -1,9 +1,16 @@
-<cd-info-panel *ngIf="true"
+<cd-info-panel *ngIf="notConfigured"
+               title="Ganesha not configured"
```
Comment (LenzGr) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.html]:
```suggestion
               title="NFS Ganesha not configured"
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.html [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+               i18n-title
+               i18n>Ganesha config location is not configured.
```
Comment (LenzGr) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.html]:
```suggestion
               i18n>The NFS Ganesha configuration location is not configured.
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.html [hunk 3]
```diff
@@ -1,9 +1,16 @@
-<cd-info-panel *ngIf="true"
+<cd-info-panel *ngIf="notConfigured"
+               title="NFS Ganesha not configured"
+               i18n-title
+               i18n>The NFS Ganesha configuration location is not configured.
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.html]:
```suggestion
               i18n>The location of the NFS Ganesha configuration is not configured.
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.ts [hunk 4]
```diff
@@ -150,6 +151,11 @@ export class NfsListComponent implements OnInit, OnDestroy {
         );
       },
       (error) => {
+        if (error.error.detail.includes('GANESHA_RADOS_POOL_NAMESPACE')) {
+          error.preventDefault();
+          this.notConfigured = true;
+        }
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.ts]: Could you please add some test?

As error testing can be somewhat tricky I wrote 2 example tests in advanced.

```diff
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.spec.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.spec.ts
index 056e6b45b0..3d9c5687b0 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.spec.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.spec.ts
@@ -5,7 +5,7 @@ import { RouterTestingModule } from '@angular/router/testing';
 
 import { ToastModule } from 'ng2-toastr';
 import { TabsModule } from 'ngx-bootstrap/tabs';
-import { BehaviorSubject, of } from 'rxjs';
+import { BehaviorSubject, of, throwError } from 'rxjs';
 
 import {
   configureTestBed,
@@ -63,6 +63,37 @@ describe('NfsListComponent', () => {
     expect(component).toBeTruthy();
   });
 
+  describe('fetch errors', () => {
+    let resp;
+    let stoppedNotification;
+
+    const testError = (errorDetail, notConfigured) => {
+      resp.error.detail = errorDetail;
+      fixture.detectChanges();
+      expect(component.notConfigured).toBe(notConfigured);
+      expect(stoppedNotification).toBe(notConfigured);
+      expect(component.onFetchError).toHaveBeenCalled();
+    };
+
+    beforeEach(() => {
+      spyOn(component, 'onFetchError');
+      stoppedNotification = false;
+      resp = {
+        error: { detail: '' },
+        preventDefault: () => (stoppedNotification = true)
+      };
+      spyOn(nfsService, 'daemon').and.callFake(() => throwError(resp));
+    });
+
+    it('test normal error', () => {
+      testError('Something gone wrong', false);
+    });
+
+    it('test component not configured error', () => {
+      testError('Oh no GANESHA_RADOS_POOL_NAMESPACE is not set', true);
+    });
+  });
+
   describe('after ngOnInit', () => {
     beforeEach(() => {
       fixture.detectChanges();
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.html (hunk 3 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.html]: Text was removed.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts [hunk 5]
```diff
@@ -164,7 +164,7 @@ export class NotificationService {
         this.taskMessageService.getErrorMessage(finishedTask)
       );
     }
-    this.show(notification);
+    return this.show(notification);
```
[the remark this answers is not in the document]
Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts]: Change was reverted.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/services/api-interceptor.service.ts [hunk 6]
```diff
@@ -50,7 +51,7 @@ export class ApiInterceptorService implements HttpInterceptor {
 
               finishedTask.success = false;
               finishedTask.exception = resp.error;
-              this.notificationService.notifyTask(finishedTask);
+              timeoutId = this.notificationService.notifyTask(finishedTask);
```
[the remark this answers is not in the document]
Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/services/api-interceptor.service.ts]: Could you create a issue suggesting this? if it still applies to master.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.ts (hunk 4 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.ts]: Change was reverted.

**Comment by tspmelo** [comment]
I have applied the suggestion from @votdev.
We now have a status endpoint and it is called during the routing phase.
This also fixes the issue I had reported today.

Currently still working on qa tests.

**Comment by tspmelo** [comment]
QA are now working, but I would accept a better way to accomplish testing the invalid status.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-501/nfs-501.component.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import { SummaryService } from '../../../shared/services/summary.service';
+
+@Component({
+  selector: 'cd-nfs-501',
+  templateUrl: './nfs-501.component.html',
+  styleUrls: ['./nfs-501.component.scss']
+})
+export class Nfs501Component implements OnInit, OnDestroy {
+  docsUrl: string;
+  message = 'The Object Gateway service is not configured.';
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-501/nfs-501.component.ts]:
```suggestion
  message = 'The NFS service is not configured.';
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/services/api-interceptor.service.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (Devp00l) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/services/api-interceptor.service.ts]: Sure created the [issue](https://tracker.ceph.com/issues/38418).

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-501/nfs-501.component.ts (hunk 7 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-501/nfs-501.component.ts]: Done

**Comment by rjfd** [comment]
QA Tests results green: http://pulpito.ceph.com/rdias-2019-02-22_22:41:49-rados:mgr-wip-rdias-testing-distro-basic-smithi

**Pull request merged**